flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -132,4 +135,60 @@
 	 */
 	boolean tableExists(ObjectPath objectPath) throws CatalogException;
 
+	// ------ partitions ------
+
+	/**
+	 * Get CatalogPartitionSpec of all partitions of the table.
+	 *
+	 * @param tablePath	path of the table
+	 * @return a list of CatalogPartitionSpec of the table
+	 *
+	 * @throws TableNotExistException thrown if the table does not exist in the catalog
+	 * @throws TableNotPartitionedException thrown if the table is not partitioned
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath)
+		throws TableNotExistException, TableNotPartitionedException, CatalogException;
+
+	/**
+	 * Get CatalogPartitionSpec of all partitions that is under the given CatalogPartitionSpec in the table.
+	 *
+	 * @param tablePath	path of the table
+	 * @param partitionSpec the partition spec to list
+	 * @return a list of CatalogPartitionSpec that is under the given CatalogPartitionSpec in the table
+	 *
+	 * @throws TableNotExistException thrown if the table does not exist in the catalog
+	 * @throws TableNotPartitionedException thrown if the table is not partitioned
+	 * @throws PartitionSpecInvalidException thrown if the given partition spec is invalid
+	 * @throws CatalogException in case of any runtime exception
+	 */
+	List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
+		throws TableNotExistException, TableNotPartitionedException, PartitionSpecInvalidException, CatalogException;
+
+	/**
+	 * Get a partition of the given table.
+	 * The given partition spec keys and values need to be matched exactly for a result.
+	 *
+	 * @param tablePath path of the table
+	 * @param partitionSpec partition spec of partition to get
+	 * @return the requested partition
+	 *
+	 * @throws TableNotExistException thrown if the table does not exist in the catalog
+	 * @throws TableNotPartitionedException thrown if the table is not partitioned
+	 * @throws PartitionSpecInvalidException thrown if the given partition spec is invalid,
+	 * @throws PartitionNotExistException thrown if the partition is not partitioned
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	CatalogPartition getPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
```

Comment:
This functionality can be covered by "listPartitions"?

Reply:
not really. PartitionSpec in listPartition can be a partial spec to indicate a subset of all partitions in the table, while it must be a full spec in getPartition to represent a single partition.

E.g., say we have a table with partition keys(k1, k2) and 2 partitions (k1=a, k2=b) and (k1=a, k2=c), `listPartitions(tablePath, (k1=a))` returns both partitions, and of course as an extreme case, `listPartitions(tablePath, (k1=a, k2=b))` returns a list with the first partition as the only element inside. `getPartition` requires a full spec not partial ones, e.g. you can pass (k1=a, k2=b), but not just (k1=a) in which case it throws exception